### PR TITLE
feat(groq-adapter): add groq adapter integration

### PR DESCRIPTION
**Pull request by abughali**
### Which issue(s) does this pull-request address?

Closes: #

### Description

- Integrated Groq's LLM capabilities with the framework through the `GroqChatLLM` adapter.
- Added support for streaming and non-streaming language model operations.
- Implemented configurable parameters such as temperature, max tokens, and top_p.
- Enabled serialization and deserialization for state persistence.
- Added usage example and a readme update.

### Checklist

- [x] I have read the [contributor guide](https://github.com/i-am-bee/bee-agent-framework/blob/main/CONTRIBUTING.md)
- [x] Linting passes: `yarn lint` or `yarn lint:fix`
- [x] Formatting is applied: `yarn format` or `yarn format:fix`
- [x] Unit tests pass: `yarn test:unit`
- [x] E2E tests pass: `yarn test:e2e`
- [x] Tests are included 
- [x] Documentation is changed or added
- [x] Commit messages and PR title follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)

### CODE REVIEW

File: examples/llms/providers/groq.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  },
+});
+
+console.info("Meta", await llm.meta());
+const response = await llm.stream([
+  BaseMessage.of({
+    role: "user",
+    text: "Hello world!",
+  }),
+]);
```
Comment (Tomas2D) [inline examples/llms/providers/groq.ts:20]: Use the `generate` method instead to be consistent with other examples

File: package.json [hunk 2]
```diff
@@ -87,6 +87,7 @@
     "dirty-json": "0.9.2",
     "duck-duck-scrape": "^2.2.5",
     "fast-xml-parser": "^4.4.1",
+    "groq-sdk": "^0.7.0",
```
[the remark this answers is not in the document]
Comment (abughali) [reply package.json]: fixed

File: examples/llms/providers/groq.ts (hunk 1 above)
Comment (abughali) [reply examples/llms/providers/groq.ts]: this is done.

**Comment by abughali** [comment]
@Tomas2D all concerned were addressed. Thanks.

**Review by Tomas2D — APPROVED** [review]

**Pull request merged**